### PR TITLE
[MM-823]: Added testcase for post.go

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
### Summary
Added testcase for server/post.go

### CODE REVIEW

File: server/post_test.go [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+				testJoinURL := "testJoinURL"
+				client.On("CreateMeeting").Return(&msgraph.OnlineMeeting{JoinURL: &testJoinURL}, nil)
+			},
+		},
+		{
+			name:          "Meeting posted successfully",
+			creator:       &model.User{Id: "testUserID", Username: "testUsername"},
+			channelID:     "testChannelID",
+			topic:         "testTopic",
+			expectedError: "",
```
Comment (raghavaggarwal2308) [inline server/post_test.go]: can remove

File: server/post_test.go [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+		channelID     string
+		topic         string
+		expectedError string
+		setup         func()
+	}{
+		{
+			name:          "User not connected",
+			creator:       &model.User{Id: "testUserID"},
+			channelID:     "testChannelID",
+			topic:         "testTopic",
```
Comment (raghavaggarwal2308) [inline server/post_test.go]: These are never changing, not need to pass here

File: server/post_test.go [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+			expectedError: "",
+			setup: func() {
+				info := &UserInfo{
+					Email: "testEmail",
+				}
+				p.setConfiguration(&configuration{
+					EncryptionKey: "demo_encrypt_key",
+				})
+				key := []byte(p.getConfiguration().EncryptionKey)
+				encryptedUserInfo, err := info.EncryptedJSON(key)
```
Comment (raghavaggarwal2308) [inline server/post_test.go]: Never changing, can be at the bottom in common part

File: server/post_test.go [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+				require.NoError(t, err)
+				api.On("KVGet", tokenKey+"testUserID").Return(encryptedUserInfo, nil).Once()
+				api.On("HasPermissionToChannel", "testUserID", "testChannelID", model.PermissionCreatePost).Return(true)
+				api.On("GetChannel", "testChannelID").Return(&model.Channel{Id: "testChannelID", Type: model.ChannelTypeDirect}, nil)
+				api.On("GetChannelMembers", "testChannelID", 0, 100).Return(model.ChannelMembers{}, nil)
+				api.On("CreatePost", mock.Anything).Return(&model.Post{}, nil)
+				testJoinURL := "testJoinURL"
+				client.On("CreateMeeting").Return(&msgraph.OnlineMeeting{JoinURL: &testJoinURL}, nil)
```
Comment (raghavaggarwal2308) [inline server/post_test.go]:
```suggestion
				client.On("CreateMeeting").Return(&msgraph.OnlineMeeting{JoinURL: mode.GetString("testJoinURL")}, nil)
```

File: server/post_test.go [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			api.ExpectedCalls = nil
+			client.ExpectedCalls = nil
+
+			tt.setup()
+
+			_, _, err := p.postMeeting(tt.creator, tt.channelID, tt.topic)
```
Comment (raghavaggarwal2308) [inline server/post_test.go]: We should test the first two params as well if possible

File: server/post_test.go [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+				testJoinURL := "testJoinURL"
+				client.On("CreateMeeting").Return(&msgraph.OnlineMeeting{JoinURL: &testJoinURL}, nil)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			api.ExpectedCalls = nil
+			client.ExpectedCalls = nil
```
Comment (raghavaggarwal2308) [inline server/post_test.go:126]: What is the use of these?

File: server/post_test.go [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+}
+
+func TestPostConnect(t *testing.T) {
+	api := &plugintest.API{}
+	p := &Plugin{
+		MattermostPlugin: plugin.MattermostPlugin{
+			API: api,
+		},
+		botUserID: "testBostUserID",
+	}
```
Comment (raghavaggarwal2308) [inline server/post_test.go]: Should not be on global level, this can lead to API being using mocks from other testcases

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: server/post_test.go (hunk 5 above)
Comment (Kshitij-Katiyar) [reply server/post_test.go]: the post and meeting returned by the function are mocked by us only hence no point in testing them

File: server/post_test.go (hunk 7 above)
Comment (Kshitij-Katiyar) [reply server/post_test.go]: Discussed

**Review by raghavaggarwal2308 — APPROVED** [review]

**Comment by Kshitij-Katiyar** [comment]
Created a PR on MM remote with all these changes, hence closing this one

**Pull request closed without merging**